src/rle.js

```diff
@@ -18,8 +18,27 @@
  */
 
 function rle(source) {
-  // write code here
+  let counter = 0;
+  let result = '';
 
+  for (let i = 0; i < source.length; i++) {
+    let current = source[i];
+    let next = source[i + 1];
+
+    if (current === next) {
+      counter++;
+    } else {
+      result += current;
+
+      if (counter > 0) {
+        result += counter + 1;
```

Comment:
maybe string interpolation here? It's not obvious that you are concatenating string and int.

Reply:
@roman-mate I've copypasted this solution from one of the students just for test. But thank you, this line is not obvious in general.